pytensor/graph/basic.py

```diff
@@ -907,6 +907,62 @@ def graph_inputs(
     yield from (r for r in ancestors(graphs, blockers) if r.owner is None)
 
 
+def required_graph_inputs(
+    graph: Variable[Any, Any] | Iterable[Variable[Any, Any]],
+) -> list[Variable[Any, Any]]:
+    """
+    Get the inputs into PyTensor variables
+
+    Parameters
+    ----------
+        graph: PyTensor `Variable` instances
+        Output `Variable` instances from which to search backward through
+        owners.
+
+    Returns
+    -------
+        List of tensor variables that are input nodes with no owner, in the order
+        found by a left-recursive depth-first search started at the nodes in `graphs`.
+
+    Examples
```

Comment:
The example is very very wrong :)

Reply:
Extremely sorry. For some reason the changes that I did got stashed and the wrong example got pushed. I'll rectify this.